### PR TITLE
feat: add column level tag support

### PR DESCRIPTION
**Pull request by jczhong84**
This PR will add column level tag support. For now, it will only show the tags synced from metastore, but not allow users to add/edit a tag through UI

Snip20230215_11

### CODE REVIEW

File: querybook/server/lib/metastore/base_metastore_loader.py [hunk 1]
```diff
@@ -329,24 +329,34 @@ def _create_table_table(
             )
 
             for column in columns:
-                create_column(
+                column_id = create_column(
```
Comment (czgu) [inline querybook/server/lib/metastore/base_metastore_loader.py:332]: just to double check, i thought flushing the row would generate the id?

Comment (czgu) [reply querybook/server/lib/metastore/base_metastore_loader.py:332]: https://stackoverflow.com/questions/28242523/get-inserted-key-before-commit-session

File: querybook/server/logic/tag.py [hunk 2]
```diff
@@ -16,6 +16,17 @@ def get_tag_by_table_id(table_id, session=None):
     )
 
 
+@with_session
+def get_tag_by_column_id(column_id: int, session=None):
```
Comment (czgu) [inline querybook/server/logic/tag.py:20]: get_tags

File: querybook/server/logic/tag.py [hunk 3]
```diff
@@ -128,3 +139,41 @@ def create_table_tags(
         session.commit()
     else:
         session.flush()
+
+
+@with_session
+def create_column_tags(
+    column_id: int = None,
```
Comment (czgu) [inline querybook/server/logic/tag.py:146]: column_id should be required

File: querybook/webapp/components/DataTableViewColumn/DataTableColumnCard.tsx [hunk 4]
```diff
@@ -69,6 +86,11 @@ export const DataTableColumnCard: React.FunctionComponent<IProps> = ({
                                 />
                             </KeyContentDisplay>
                         )}
+                        {columnTags && columnTags.length > 0 && (
+                            <KeyContentDisplay keyString="Tags">
```
Comment (czgu) [inline querybook/webapp/components/DataTableViewColumn/DataTableColumnCard.tsx:90]: better if you do
```
    const tagsDOM = (columnTags || []).map((tag) => (
                <TableTag tag={tag} readonly={true} key={tag.id} mini={true} />
            ));

   tagsDOM && (
      <KeyContentDisplay keyString="Tags"><div className="DataTableTags flex-row">{tagsDOM}</div>
      </KeyContentDisplay>

  )

```

Comment (jczhong84) [reply querybook/webapp/components/DataTableViewColumn/DataTableColumnCard.tsx:90]: tagsDom is not falsey when columnTags is []

File: querybook/server/lib/metastore/base_metastore_loader.py (hunk 1 above)
Comment (jczhong84) [reply querybook/server/lib/metastore/base_metastore_loader.py:332]: yep, flush actually works

File: querybook/webapp/components/DataTableViewColumn/DataTableColumnCard.tsx (hunk 4 above)
Comment (czgu) [reply querybook/webapp/components/DataTableViewColumn/DataTableColumnCard.tsx:90]: oh then tagsDOm.length

File: querybook/server/datasources/tag.py [hunk 5]
```diff
@@ -11,10 +12,19 @@
...
+    verify_data_table_permission(table_id)
+    return logic.get_tags_by_table_id(table_id=table_id)
+
+
+@register(
+    "/column/<int:column_id>/tag/",
+    methods=["GET"],
+)
+def get_tags_by_column_id(column_id: int):
+    column = metastore_logic.get_column_by_id(column_id)
```
Comment (czgu) [inline querybook/server/datasources/tag.py:25]: can you add `verify_data_column_permission` so the checks are in 1 place?

**Review by czgu — APPROVED** [review]

**Pull request merged**